corehq/apps/app_manager/models.py

```diff
@@ -4788,27 +4791,32 @@ def _hash(val):
                 else:
                     form.version = None
 
-    def set_media_versions(self):
+    def set_media_versions(self, version_reverted_to=None):
         """
         Set the media version numbers for all media in the app to the current app version
-        if the media is new or has changed since the last build. Otherwise set it to the
-        version from the last build.
+        if the media is new or has changed since the last build or the build reverting to.
+        Otherwise set it to the version from the last build/version reverted to.
         """
+        icds_toggle_enabled = toggles.ICDS.enabled(self.domain)
 
-        # access to .multimedia_map is slow
-        previous_version = self.get_previous_version()
+        if version_reverted_to and icds_toggle_enabled:
+            previous_version = wrap_app(get_build_doc_by_version(self.domain, self.copy_of, version_reverted_to))
+        else:
+            previous_version = self.get_previous_version()
         prev_multimedia_map = previous_version.multimedia_map if previous_version else {}
-
+        is_linked_app = isinstance(self, LinkedApplication)
         for path, map_item in six.iteritems(self.multimedia_map):
             prev_map_item = prev_multimedia_map.get(path, None)
             if prev_map_item and prev_map_item.unique_id:
                 # Re-use the id so CommCare knows it's the same resource
                 map_item.unique_id = prev_map_item.unique_id
-            if (prev_map_item and prev_map_item.version
-                    and prev_map_item.multimedia_id == map_item.multimedia_id):
-                map_item.version = prev_map_item.version
-            else:
-                map_item.version = self.version
+            # if its a linked app and the map item already has a version don't need to set it up again
+            if not (icds_toggle_enabled and is_linked_app and map_item.version):
```

Comment:
i think map_item.version would always be present for a linked app, but i have added this as a safe keeping for cases i can't think of

Reply:
This block will also never get executed on ICDS which means `map_item.version` will never get set, is that correct?

Reply:
If `map_item.version` isn't present for a linked app this would mean it never get's set - is that an issue?

Reply:
@snopoke thanks for the feedback 
1. this block should not get executed when 
- the toggle is enabled and 
- is a linked app and 
- the version is already set. 
So on ICDS it would run if its not a linked app or if its a linked app but map_item.version is not set(which i said i am not sure is possible since the master app would always set it)
2. if map_item.version is not set this would run.